roles/aci/tasks/main.yml

```diff
@@ -18,11 +18,14 @@
     name: "{{ item }}"
   with_items: "{{ annotate_namespace_list }}"
   delegate_to: "{{ groups.oo_first_master.0 }}"
-- name: Get aci policy tenant name
-  shell: "grep -m1 'aci-policy-tenant' {{ aci_deployment_yaml_file }} | awk -F '\"' '{print $4}'"
-  register: tenant
-  delegate_to: "{{ groups.oo_first_master.0 }}"
+- name: Copy annotate_namespace.sh to /tmp
+  copy:
+    src: annotate_namespace.sh
+    dest: /tmp/annotate_namespace.sh
+    owner: root
+    group: root
+    mode: 0777
 - name: Annotate namespaces created
-  shell: "oc annotate namespace {{ item }} opflex.cisco.com/endpoint-group='{\"policy-space\":\"{{ tenant.stdout }}\", \"name\": \"kubernetes|kube-system\"}' --overwrite=True"
+  command: sh /tmp/annotate_namespace.sh {{ item }} {{ aci_deployment_yaml_file }}
```

Comment:
Thats a good suggestion. We wanted to avoid complicated logic in ansible playbook which involves " and '.
Putting those commands into a script and then running that script seems to be lot easier to maintain and easy to read over doing escape char business.
We ultimately want to execute this command
`kubectl annotate namespace {{ namespace }} opflex.cisco.com/endpoint-group='{"policy-space":"{{ tenant_name }}", "name": "kubernetes|kube-system"}' --overwrite=True`

I am seeing similar pattern here:
https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/openshift_hosted/tasks/storage/glusterfs.yml#L29
https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/calico/templates/calicov3.yml.j2#L277
https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/nuage_master/templates/nuage-infra-pod-config-daemonset.j2#L52

Reply:
@patrickdillon : Do you have any comment on this one?

Reply:
@patrickdillon : Gentle reminder

Reply:
@gaurav-dalvi I think you can mitigate the confusion of the quotation marks by not including them at the beginning of the line. As a suggestion, based on your comment, something like this would be preferable:
```
command: >
  oc annotate namespace {{ item }} opflex.cisco.com/endpoint-group='{"policy-space":"{{ tenant.stdout }}", "name": "kubernetes|kube-system"}' --overwrite=True
```
For the tenant `my_tenant`, the annotation looks like this: `opflex.cisco.com/endpoint-group={"policy-space":"my_tenant", "name": "kubernetes|kube-system"}`

We want to avoid the pattern of copying a script to the host and executing it because Ansible is already built to handle that use case with the command module. The first link you sent is just creating a temp directory, which is fine in and of itself, especially because that temp dir is needed by gluster. The second two links are pod templates. Those scripts are not copied or executed by Ansible. Let me know if this explanation is confusing.

Reply:
Thanks @patrickdillon . I have done the necessary changes. Could you please take a look now? 
At line 22 in main.yml: same command using `command` module did not work hence I am using shell module.